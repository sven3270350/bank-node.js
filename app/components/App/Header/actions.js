/*
 *
 * HeaderApp actions
 *
 */

import {
  TOGGLE_MENU_DESKTOP,
  TOGGLE_MENU_MOBILE,
  LOGOUT,
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  IS_NOTIFICATION,
  NEW_NOTIFICATION,
  UNSET_NOTIFICATION,
  HIDDEN_MENU_MOBILE,
  TOGGLE_NOTIFICATION,
} from './constants';

export function toggleNotificationAction() {
  return {
    type: TOGGLE_NOTIFICATION,
  };
}

export function isNotificationAction() {
  return {
    type: IS_NOTIFICATION,
  };
}

export function newNotificationAction(notificationCount) {
  return {
    type: NEW_NOTIFICATION,
    notificationCount,
  };
}

export function unsetNotificationAction() {
  return {
    type: UNSET_NOTIFICATION,
  };
}

export function toggleMenuDesktopAction() {
  return {
    type: TOGGLE_MENU_DESKTOP,
  };
}

export function toggleMenuMobileAction() {
  return {
    type: TOGGLE_MENU_MOBILE,
  };
}

export function hiddenMenuMobileAction() {
  return {
    type: HIDDEN_MENU_MOBILE,
  };
}

export function logoutAction(id) {
  return {
    type: LOGOUT,
    id,
  };
}

export function successLogoutAction() {
  localStorage.removeItem('id_token');
  return {
    type: LOGOUT_SUCCESS,
  };
}

export function errorLogoutAction(error) {
  return {
    type: LOGOUT_ERROR,
    error,
  };
}
