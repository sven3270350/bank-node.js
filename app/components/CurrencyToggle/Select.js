import styled from 'styled-components';

const Select = styled.select`
  line-height: 1em;
  height: 20px;
  border: 1px solid grey;
  height: 36px;
  width: 17rem;
  border-radius: 2px;
  -webkit-appearance: none;
  background-color: white;
  padding: 10px;

  &:hover {
    cursor: pointer;
  }
`;

export default Select;
