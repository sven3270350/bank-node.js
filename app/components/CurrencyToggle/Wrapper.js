import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 0px;
`;

export default Wrapper;
