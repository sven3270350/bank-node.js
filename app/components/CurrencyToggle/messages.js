/*
 * CurrencyToggle Messages
 *
 * This contains all the text for the CurrencyToggle component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.CurrencyToggle';

export default defineMessages({
  1: {
    id: `${scope}.PLN`,
    defaultMessage: 'PLN',
  },
  2: {
    id: `${scope}.USD`,
    defaultMessage: 'USD',
  },
  3: {
    id: `${scope}.EUR`,
    defaultMessage: 'EUR',
  },
});
