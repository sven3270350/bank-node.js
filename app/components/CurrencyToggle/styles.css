.select__currency {
  position: relative;
  display: inline-block;
  width: 17rem;
}
.select__currency select {
  font-family: 'Lato';
  display: inline-block;
  width: 100%;
  cursor: pointer;
  padding: 8px 10px;
  outline: 0;
  border: 1px solid grey;
  border-radius: 2px;
  background: #ffffff;
  color: #000000;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
}

.select__currency select::-ms-expand {
  display: none;
}

.select__currency select:hover,
.select__currency select:focus {
  color: #000000;
  background: #ffffff;
}

.select__currency select:disabled {
  opacity: 1;
  pointer-events: none;
}

.select_arrow--currency {
  position: absolute;
  top: 11px;
  right: 15px;
  width: 10px;
  height: 10px;
  border: solid #0029ab;
  border-width: 0 2px 2px 0;
  display: inline-block;
  padding: 3px;
  transform: rotate(45deg);
  -webkit-transform: rotate(45deg);
  transition: 0.3s ease;
}

.select__currency select:hover ~ .select_arrow--currency,
.select__currency select:focus ~ .select_arrow--currency {
  border-color: #0029ab;
}

.select__currency select:disabled ~ .select_arrow--currency {
  border-top-color: #0029ab;
}

.select__currency select:focus ~ .select_arrow--currency {
  transform: translateY(40%) rotate(225deg);
}

.select__currency select:not(:hover) ~ .select_arrow--currency {
  transform: rotate(45deg);
}

@media screen and (max-width: 480px) {
  .select__currency {
    position: relative;
    display: inline-block;
    width: 100%;
  }
}
