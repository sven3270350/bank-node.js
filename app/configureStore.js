/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { fromJS } from 'immutable';
import { routerMiddleware } from 'connected-react-router/immutable';
import createSagaMiddleware from 'redux-saga';
import immutableTransform from 'redux-persist-transform-immutable';
import storage from 'redux-persist/lib/storage';
import { persistReducer } from 'redux-persist';
import { persistStore, autoRehydrate } from 'redux-persist-immutable';
import createReducer from './reducers';

const persistConfig = {
  transforms: [immutableTransform()],
  key: 'root',
  storage,
};

const sagaMiddleware = createSagaMiddleware();
const rootReducer = createReducer();
const persistedReducer = persistReducer(persistConfig, rootReducer);

export default function configureStore(initialState = {}, history) {
  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares), autoRehydrate()];

  // If Redux DevTools Extension is installed use it, otherwise use Redux compose
  /* eslint-disable no-underscore-dangle, indent */
  const composeEnhancers =
    process.env.NODE_ENV !== 'production' &&
    typeof window === 'object' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose;
  /* eslint-enable */

  const store = createStore(
    persistedReducer,
    fromJS(initialState),
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers));
    });
  }

  const persistor = persistStore(store);
  return { store, persistor };
}
