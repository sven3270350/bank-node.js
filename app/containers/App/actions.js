/*
 *
 * App actions
 *
 */

import {
  IS_LOGGED,
  LOGGED_IN,
  LOGOUT,
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  ENQUEUE_SNACKBAR,
  CLOSE_SNACKBAR,
  REMOVE_SNACKBAR,
  TOGGLE_NAVIGATION_DESKTOP,
  TOGGLE_NAVIGATION_MOBILE,
  TOGGLE_MESSAGES,
  TOGGLE_NOTIFICATIONS,
  CHECK_NEW_MESSAGES,
  CHECK_NEW_MESSAGES_SUCCESS,
  CHECK_NEW_MESSAGES_ERROR,
  CHECK_NEW_NOTIFICATIONS,
  CHECK_NEW_NOTIFICATIONS_SUCCESS,
  CHECK_NEW_NOTIFICATIONS_ERROR,
  GET_NEW_NOTIFICATIONS,
  GET_NEW_NOTIFICATIONS_SUCCESS,
  GET_NEW_NOTIFICATIONS_ERROR,
  UNSET_NEW_NOTIFICATIONS,
  UNSET_NEW_NOTIFICATIONS_SUCCESS,
  UNSET_NEW_NOTIFICATIONS_ERROR,
} from './constants';

export function checkNewMessagesAction() {
  return {
    type: CHECK_NEW_MESSAGES,
  };
}

export function checkNewMessagesSuccessAction(messages) {
  return {
    type: CHECK_NEW_MESSAGES_SUCCESS,
    messages,
  };
}

export function checkNewMessagesErrorAction(error) {
  return {
    type: CHECK_NEW_MESSAGES_ERROR,
    error,
  };
}

export function checkNewNotificationsAction() {
  return {
    type: CHECK_NEW_NOTIFICATIONS,
  };
}

export function checkNewNotificationsSuccessAction(notificationCount) {
  return {
    type: CHECK_NEW_NOTIFICATIONS_SUCCESS,
    notificationCount,
  };
}

export function checkNewNotificationsErrorAction(error) {
  return {
    type: CHECK_NEW_NOTIFICATIONS_ERROR,
    error,
  };
}

export function getNewNotificationsAction() {
  return {
    type: GET_NEW_NOTIFICATIONS,
  };
}

export function getNewNotificationsSuccessAction(notifications) {
  return {
    type: GET_NEW_NOTIFICATIONS_SUCCESS,
    notifications,
  };
}

export function getNewNotificationsErrorAction(error) {
  return {
    type: GET_NEW_NOTIFICATIONS_ERROR,
    error,
  };
}

export function unsetNewNotificationsAction() {
  return {
    type: UNSET_NEW_NOTIFICATIONS,
  };
}

export function unsetNewNotificationsSuccessAction() {
  return {
    type: UNSET_NEW_NOTIFICATIONS_SUCCESS,
  };
}

export function unsetNewNotificationsErrorAction(error) {
  return {
    type: UNSET_NEW_NOTIFICATIONS_ERROR,
    error,
  };
}

export function isLoggedAction() {
  return {
    type: IS_LOGGED,
  };
}

export function loggedInAction(userId, token) {
  return {
    type: LOGGED_IN,
    userId,
    token,
  };
}

export function logoutAction() {
  return {
    type: LOGOUT,
  };
}

export function logoutSuccessAction() {
  return {
    type: LOGOUT_SUCCESS,
  };
}

export function toggleNavigationDesktopAction() {
  return {
    type: TOGGLE_NAVIGATION_DESKTOP,
  };
}

export function toggleNavigationMobileAction() {
  return {
    type: TOGGLE_NAVIGATION_MOBILE,
  };
}

export function toggleMessagesAction() {
  return {
    type: TOGGLE_MESSAGES,
  };
}

export function toggleNotificationsAction() {
  return {
    type: TOGGLE_NOTIFICATIONS,
  };
}

export function logoutErrorAction(error) {
  return {
    type: LOGOUT_ERROR,
    error,
  };
}

export function enqueueSnackbarAction(snackbar) {
  const key = snackbar.options && snackbar.options.key;

  return {
    type: ENQUEUE_SNACKBAR,
    snackbar: {
      ...snackbar,
      key: key || new Date().getTime() + Math.random(),
    },
  };
}

export function closeSnackbarAction(key) {
  return {
    type: CLOSE_SNACKBAR,
    dismissAll: !key,
    key,
  };
}

export function removeSnackbarAction(key) {
  return {
    type: REMOVE_SNACKBAR,
    key,
  };
}
