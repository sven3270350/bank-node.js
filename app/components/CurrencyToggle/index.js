/* eslint-disable radix */
/**
 *
 * CurrencyToggle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import './styles.css';

// Import Components
import { makeCurrencySelector } from 'containers/RegisterPage/selectors';
import Toggle from './Toggle';
import messages from './messages';

import { changeCurrencyAction } from '../../containers/RegisterPage/actions';

class CurrencyToggle extends React.PureComponent {
  render() {
    return (
      <Toggle
        value={this.props.currencyId}
        values={this.props.currency}
        messages={messages}
        onToggle={this.props.onCurrencyToggle}
      />
    );
  }
}

CurrencyToggle.propTypes = {
  currencyId: PropTypes.number,
  currency: PropTypes.array,
  onCurrencyToggle: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  currency: makeCurrencySelector(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onCurrencyToggle: evt =>
      dispatch(changeCurrencyAction(parseInt(evt.target.value))),
    dispatch,
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CurrencyToggle);
