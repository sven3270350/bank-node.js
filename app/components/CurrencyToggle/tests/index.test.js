import React from 'react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from '@testing-library/react';

import { changeCurrencyAction } from 'containers/RegisterPage/actions';
import CurrencyToggle, { mapDispatchToProps } from '../index';

import configureStore from '../../../configureStore';

describe('<CurrencyToggle />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <CurrencyToggle />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should present the default `en` english language option', () => {
    const { container } = render(
      <Provider store={store}>
        <CurrencyToggle />
      </Provider>,
    );
    expect(container.querySelector('option[value="en"]')).not.toBeNull();
  });

  describe('mapDispatchToProps', () => {
    describe('onLocaleToggle', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLocaleToggle).toBeDefined();
      });

      it('should dispatch changeLocale when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const currency = 1;
        const evt = { target: { value: currency } };
        result.onLocaleToggle(evt);
        expect(dispatch).toHaveBeenCalledWith(changeCurrencyAction(currency));
      });
    });
  });
});
