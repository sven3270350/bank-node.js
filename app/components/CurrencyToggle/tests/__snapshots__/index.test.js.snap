// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyToggle /> should match the snapshot 1`] = `
<div
  class="SelectWrapper-ftlc34-0 eDeigd"
>
  <select
    class="Select-fhpbt0-0 cgpFOs"
  />
  <div
    class="Arrow-sc-172g500-0 jcSFyu"
  />
</div>
`;
